tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,460 @@
+	hubIPv4ExternalAddresses []string,
+	ipv4NodeAddrList []string,
+	hubPodWorkerName []string,
+	containerFrrName string,
+	workerNodeList []*nodes.Builder,
+	masterNodeList []*nodes.Builder) *pod.Builder {
+	var (
+		externalMacVlanNADName = "external"
+		nadHubName             = "nad-hub"
+		masterPodIPv4Prefix    = "172.16.0.1/24"
+	)
+
+	By("Creating External NAD for master FRR pod")
+	createExternalNad(externalMacVlanNADName)
+
+	By("Creating External NAD for hub FRR pods")
+	createExternalNad(nadHubName)
+
+	By("Creating static ip annotation for hub0")
+
+	hub0BRstaticIPAnnotation := createStaticIPAnnotations(externalMacVlanNADName,
+		nadHubName,
+		[]string{fmt.Sprintf("%s/24", ipv4SecurityIPList[0])},
+		[]string{fmt.Sprintf("%s/24", hubIPv4ExternalAddresses[0])})
+
+	By("Creating static ip annotation for hub1")
+
+	hub1BRstaticIPAnnotation := createStaticIPAnnotations(externalMacVlanNADName, nadHubName,
+		[]string{fmt.Sprintf("%s/24", ipv4SecurityIPList[1])},
+		[]string{fmt.Sprintf("%s/24", hubIPv4ExternalAddresses[1])})
+
+	By("Creating Frr Hub pod configMap")
+
+	hubConfigMap := createHubConfigMap("hub-node-config")
+
+	By("Creating FRR Hub pod on worker node 0")
+
+	_ = createFrrPodTest(hubPodWorkerName[0],
+		workerNodeList[0].Object.Name, hubConfigMap.Definition.Name, containerFrrName, []string{},
+		hub0BRstaticIPAnnotation, false)
+
+	By("Creating FRR Hub pod on worker node 1")
+
+	_ = createFrrPodTest(hubPodWorkerName[1],
+		workerNodeList[1].Object.Name, hubConfigMap.Definition.Name, containerFrrName, []string{},
+		hub1BRstaticIPAnnotation, false)
+
+	By("Creating configmap and Frr Master pod")
+
+	configMapStaticRoutes := defineConfigMapWithStaticRouteAndNetwork(hubIPv4ExternalAddresses,
+		removePrefixFromIPList(ipv4NodeAddrList))
+	masterConfigMap := createMasterConfigMap("master-configmap", configMapStaticRoutes)
+
+	masterStaticIPAnnotation := createStaticIPAnnotations(externalMacVlanNADName, nadHubName,
+		[]string{masterPodIPv4Prefix}, []string{""})
+
+	return createFrrPodTest("master-pod", masterNodeList[0].Definition.Name,
+		masterConfigMap.Definition.Name, containerFrrName, []string{}, masterStaticIPAnnotation, true)
+}
+
+func createStaticIPAnnotations(internalNADName, externalNADName string, internalIPAddresses,
+	externalIPAddresses []string) []*types.NetworkSelectionElement {
+	ipAnnotation := pod.StaticIPAnnotation(internalNADName, internalIPAddresses)
+
+	return append(ipAnnotation,
+		pod.StaticIPAnnotation(externalNADName, externalIPAddresses)...)
+}
+
+func createHubConfigMap(name string) *configmap.Builder {
+	configMapData := defineBaseConfig(tsparams.DaemonsFile, "", "")
+	hubConfigMap, err := configmap.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create hub config map")
+
+	return hubConfigMap
+}
+
+func createMasterConfigMap(name, configMapStaticRoutes string) *configmap.Builder {
```

Comment:
Please rename createHubConfigMap func and reuse it

Reply:
renamed and reused